recipe/build.sh

```diff
@@ -20,26 +21,102 @@ if [[ "$CONDA_BUILD_CROSS_COMPILATION" == "1" ]]; then
       exit 1
       ;;
   esac
-  EXTRA_CONFIGURE_OPTIONS="--enable-cross-compile --arch=$ARCH --target-os=$OS --cross-prefix=$HOST- --host-cc=$CC_FOR_BUILD"
+  extra_args="${extra_args} --enable-cross-compile --arch=$ARCH --target-os=$OS --cross-prefix=$HOST- --host-cc=$CC_FOR_BUILD"
 fi
 
-extra_args=""
-if [[ "${target_platform}" == "linux-64" ]]; then
-  extra_args=--enable-vaapi
+if [[ "${target_platform}" == "win-64" ]]; then
+  # 2022/07 hmaarrfk
+  # Specifying these extra flags for osx and linux
+  # seems to cause things to fail since FFmpeg
+  # expects cc to be the ld
+  # # https://github.com/FFmpeg/FFmpeg/blob/master/configure#L4894
+  # the LDFLAGs on Unix contain the -Wl, prefix, and thus
+  # are incompatible with the direct ld command
+  # Thus we avoid specifying ld for the unix platforms
+  extra_args="${extra_args} --ld=${LD}"
+
+  extra_args="${extra_args} --target-os=win64"
+  extra_args="${extra_args} --enable-cross-compile"
+  extra_args="${extra_args} --toolchain=msvc"
+  extra_args="${extra_args} --host-cc=${CC}"
+  extra_args="${extra_args} --extra-libs=ucrt.lib --extra-libs=vcruntime.lib --extra-libs=oldnames.lib"
+  extra_args="${extra_args} --strip=llvm-strip"
+  # Through, locally, I get
+  #    This app can't run on your PC
+  # and access denied on the terminal
+  # I cannot even run llvm-strip on the terminal
+  extra_args="${extra_args} --disable-stripping"
+
+  # ffmpeg by default attempts to link to libm
+  # but that doesn't exist for windows
+  extra_args="${extra_args} --host-extralibs="
+
+  # Delete line that includes unistd.h from zconf. we should patch this
+  # better for LLVM compatibility
+  # I submitted a ticket upstream
+  # https://github.com/madler/zlib/issues/674
+  # cp ${PREFIX}/include/zconf.h zconf.h.backup
+  # sed -i "/unistd/d" ${PREFIX}/include/zconf.h
+  if [[ ! -f "${PREFIX}/include/unistd.h" ]]; then
+      UNISTD_CREATED=1
+      touch "${PREFIX}/include/unistd.h"
+  fi
+
+  # Add pkgconfig from the prefix to search through it correctly
+  export PKG_CONFIG_PATH=${PREFIX}/lib/pkgconfig:${PKG_CONFIG_PATH}
+  PKG_CONFIG="${BUILD_PREFIX}/Library/bin/pkg-config"
+
+  # I'm not sure, but it seems like
+  # their hacky way of silencing things is messing with the AR command
+  # Especially as it is called in compat/windows/makedef
+  # https://github.com/FFmpeg/FFmpeg/blob/master/ffbuild/common.mak#L21
+  # To avoid hacky patches, I'm just going to make it verbose, always.
+  # maybe related to https://trac.ffmpeg.org/ticket/6620
+  export V=1
+  # I'm really not sure why AR is getting corupted with a
+  # @printf statement somewhere.
+  # This seemed to happen once we included `pkg-config` as a host dependency
+  # sed -i "2 aexport AR=${AR}" compat/windows/makedef
+elif [[ "${target_platform}" == linux-* ]]; then
```

Comment:
Does this comment still relate to the `export V=1` above? Normally the comments are above the line they comment on, and the ar issues seem mentioned twice

Reply:
Thank you, I cleaned up this comment and moved it above.

I rebased to keep the PR shorter. so it will appear to have been folded in the previous commit. i also backported to the 4.4.x branch PR